(backport 87637/88593 to 4.1) modules: mbedtls: update to 3.6.3 and tf-m to 2.1.2

wearyzen opened the pull request:
Update Mbed TLS to 3.6.3 as it has CVE fixes.
Fixes #88434
Related to #87637

wearyzen committed:
modules: mbedtls: update to 3.6.3

Update Mbed TLS to 3.6.3 as it has CVE fixes.

Signed-off-by: Sudan Landge <sudan.landge@arm.com>

tomi-font commented on doc/releases/release-notes-4.1.rst:
I think that we want to have the release notes for 4.1.1 separate from (on top of) the 4.1.0 ones. As I did in my 3.7 PR.

tomi-font reviewed (approved):
Minor point, I think it could be nice to have an indication in the PR title that this is a backport for 4.1.

wearyzen commented:
> Minor point, I think it could be nice to have an indication in the PR title that this is a backport for 4.1.

Done, also added a "related to" so that main PR has some info about the backports.

tomi-font reviewed (changes requested):
needs changes in TF-M as well to work properly

wearyzen committed:
modules: tf-m: update to 2.1.2

Update TF-M to 2.1.2 from version 2.1.1.
This is required to use MbedTLS 3.6.3.

Signed-off-by: Sudan Landge <sudan.landge@arm.com>

tomi-font commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -26,6 +26,47 @@
+
+Trusted Firmware-M (TF-M)
+*************************
+* Updated TF-M to version 2.1.2 (from 2.1.1). The release notes can be found at:
+  https://trustedfirmware-m.readthedocs.io/en/tf-mv2.1.2/releases/2.1.2.html
```
```suggestion
TF-M was updated to version 2.1.2 (from 2.1.1). The release notes can be found at:
https://trustedfirmware-m.readthedocs.io/en/tf-mv2.1.2/releases/2.1.2.html
```

tomi-font commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -26,6 +26,47 @@
+with security and bug fixes until at least March 2027.
+
+Trusted Firmware-M (TF-M)
+*************************
```
```suggestion
*************************

```

tomi-font reviewed (changes requested):
also suggest renaming the PR title to include information about the TF-M update

wearyzen commented:
@fabiobaltieri since you previously approved the PR could you please have a look and help with approval/merge of this?

fabiobaltieri commented:
DNM manifest is for the change in trusted-firmware-m IIUC, seems ok
